lib/modules/hosts/youtube.js

```diff
@@ -11,49 +12,48 @@ export default {
 			(/^https?:\/\/(?:youtu\.be|(?:www\.|m\.)?youtube\.com)\/?(?:watch|embed)?(?:.*?v=|v\/|\/)([\w-]+)/i).exec(href);
 	},
 	handleLink(elem, [, hash]) {
-		// Check url for timecode e.g t=1h23m15s
-		const starttimeRe = /[&\?](?:t=|start=)([0-9hms]+)/i;
-		const starttimeResult = starttimeRe.exec(elem.href);
-		// End time is always in integer format, no need to parse
-		const endtimeRe = /[&\?]end=([0-9]+)/i;
-		const endtimeResult = endtimeRe.exec(elem.href);
-		// Finds playlist id if included
-		const playlistRe = /[&\?]list=([\w\-_]+)/i;
-		const playlistResult = playlistRe.exec(elem.href);
+		let src = `https://www.youtube.com/embed/${hash}?enablejsapi=1&enablecastapi=1`;
+		const paramResults = getUrlParams(elem.href);
+		for (const paramName in paramResults) {
```

Comment:
I bet the loop is actually slower :stuck_out_tongue_winking_eye:  since V8 can optimize direct property loads like `params.t` to loads by an offset, as if it were a struct (well, ideally -- depending on object shape, inline cache state, etc.), and collecting all of an object's keys for for..in is (relatively) expensive. But performance doesn't matter here regardless.

Testing for property existence/truthiness with `if (param.t)` is very common in JS; tbh I don't think I've ever seen a for..in loop used like that before. :sweat_smile:

I agree with the duplication but we'll worry about it when more than two cases are duplicated.